### PR TITLE
Room config enhancement and bugfixes

### LINKED ISSUE #315: enable Layer 0 in Room Config Dialog cannot be applied to the current Room
caused by code changes in commit:
https://github.com/wario-land/WL4Editor/commit/080f88d3684cdfde311e8220d457381b203c538e

### PR DESCRIPTION
**Pull request by shinespeciall**
Fix #315 
now Layer 0 and Layer 2 can have data recovery when undo disable operation on them (layer 0 is a bit different and complicated, but the data recovery still work)

### CODE REVIEW

File: WL4EditorWindow.cpp [hunk 1]
```diff
@@ -546,35 +546,86 @@ void WL4EditorWindow::RoomConfigReset(DialogParams::RoomConfigParams *currentroo
     if (nextroomconfig->Layer2Enable && !currentroomconfig->Layer2Enable)
     {
         currentRoom->GetLayer(2)->CreateNewLayer_type0x10(nextroomconfig->RoomWidth, nextroomconfig->RoomHeight);
+        if(nextroomconfig->LayerData[2]) // recover layer 2 data if needed, don't need to set dirty
+        {
+            // replace layer data
+            delete[] currentRoom->GetLayer(2)->GetLayerData();
+            size_t datasize2 = 2 * nextroomconfig->RoomWidth * nextroomconfig->RoomHeight;
+            unsigned short *tmpLayerdata2 = new unsigned short[datasize2 >> 1];
+            memset(tmpLayerdata2, 0, datasize2);
```
Comment (Goldensunboy) [inline WL4EditorWindow.cpp:555]: the memset is useless

Comment (Goldensunboy) [reply WL4EditorWindow.cpp:555]: and the one in the next if clause

Comment (Goldensunboy) [reply WL4EditorWindow.cpp:555]: also lines 612 and 622

**Review by Goldensunboy — CHANGES_REQUESTED** [review]

File: WL4EditorWindow.cpp (hunk 1 above)
Comment (shinespeciall) [reply WL4EditorWindow.cpp]: done.

**Review by Goldensunboy — APPROVED** [review]

**Pull request merged**